libs/category/src/drivers/magento/transformers/applied-sort-option-transformer.service.ts

```diff
@@ -1,15 +1,25 @@
 import { Injectable } from '@angular/core';
 
-import { MagentoSortFieldAction } from '../models/requests/sort';
+import { MagentoSortFieldAction, MagentoSortDirectionEnum } from '../models/requests/sort';
+import { DaffSortDirectionEnum } from '../../../models/requests/category-request';
 
 @Injectable({
   providedIn: 'root'
 })
 export class DaffMagentoAppliedSortOptionTransformService {
 
-  transform(appliedOption: string, appliedDirection: string): MagentoSortFieldAction {
+  transform(appliedOption: string, appliedDirection: DaffSortDirectionEnum): MagentoSortFieldAction {
+		if(!appliedOption) return null;
 		return {
-			[appliedOption]: appliedDirection
+			[appliedOption]: this.transformDirection(appliedDirection)
 		}
-  }
+	}
+	
+	private transformDirection(direction: DaffSortDirectionEnum): MagentoSortDirectionEnum {
```

Comment:
What if direction is null?

Reply:
fixed. But moved this code to the category service